core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -102,7 +102,7 @@ class SocketServer(val config: KafkaConfig, val metrics: Metrics, val time: Time
             metrics.metricName("io-wait-ratio", "socket-server-metrics", p.metricTags)
           }
           ioWaitRatioMetricNames.map { metricName =>
-            Option(metrics.metric(metricName)).fold(0.0)(_.value)
+            Option(metrics.metric(metricName)).fold(0.0)(m => if (m.value > 1.0) 1.0 else m.value)
```

Comment:
nits: would it be a bit cleaner to use Math.min(m.value, 1.0)? Just for your choice. I am OK with the current approach as well.